### ISSUE TITLE
Extra depencies not being fetched?

### ISSUE THREAD

**Issue by sujaymansingh** [python-poetry/poetry#3217]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: MacOS 10.14.6
- **Poetry version**: 1.1.2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: https://gist.github.com/sujaymansingh/7be56d231276c0ed1e06ebe7a69f8b10#file-pyproject-toml

## Issue

When installing [apache-airflow](https://github.com/apache/airflow), you can specify extra dependencies (e.g. its `gcp_api` module, or to use `celery` etc)

However, when using that, it looks like poetry isn't fetching the extra dependencies.

E.g. installing `apache-airflow[celery]` should also include things like `celery` and `flower`
https://github.com/apache/airflow/blob/1.10.12/setup.py#L205-L209

However, it is not doing so.

```
$ poetry --version
Poetry version 1.1.2
$ poetry init

This command will guide you through creating your pyproject.toml config.

Package name [testing-poetry-with-tilde]:
Version [0.1.0]:
Description []:
Author [Sujay Mansingh <>, n to skip]:
License []:
Compatible Python versions [^3.9]:

Would you like to define your main dependencies interactively? (yes/no) [yes] no
Would you like to define your development dependencies interactively? (yes/no) [yes] no
Generated file

[tool.poetry]
name = "testing-poetry-with-tilde"
version = "0.1.0"
description = ""
authors = ["Sujay Mansingh <>"]

[tool.poetry.dependencies]
python = "^3.9"

[tool.poetry.dev-dependencies]

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"


Do you confirm generation? (yes/no) [yes] yes
$ poetry add 'apache-airflow[celery]'
Creating virtualenv testing-poetry-with-tilde-3DgP-Kc1-py3.9 in /Users/sujaymansingh/Library/Caches/pypoetry/virtualenvs
Using version ^1.10.12 for apache-airflow

Updating dependencies
Resolving dependencies... (15.9s)

Writing lock file

Package operations: 82 installs, 0 updates, 0 removals

  • Installing markupsafe (1.1.1)
  • Installing click (7.1.2)
  • Installing itsdangerous (1.1.0)
  • Installing jinja2 (2.11.2)
  • Installing pytz (2020.1)
  • Installing werkzeug (0.16.1)
  • Installing attrs (19.3.0): Installing...
  • Installing babel (2.8.0): Pending...
  • Installing dnspython (2.0.0): Installing...
  • Installing attrs (19.3.0)
  • Installing babel (2.8.0)
  • Installing dnspython (2.0.0)
  • Installing flask (1.1.2)
  • Installing idna (2.10)
  • Installing marshmallow (2.21.0)
  • Installing pyjwt (1.7.1)
  • Installing pyrsistent (0.17.3)
  • Installing python-openid (2.2.5)
  • Installing six (1.15.0)
  • Installing sqlalchemy (1.3.20)
  • Installing text-unidecode (1.3)
  • Installing wtforms (2.3.3)
  • Installing apispec (1.3.3): Installing...
  • Installing certifi (2020.6.20): Installing...
  • Installing chardet (3.0.4): Downloading... 30%
  • Installing colorama (0.4.4): Downloading... 0%
  • Installing docutils (0.16): Pending...
  • Installing email-validator (1.1.1): Installing...
  • Installing apispec (1.3.3)
  • Installing certifi (2020.6.20)
  • Installing chardet (3.0.4)
  • Installing colorama (0.4.4)
  • Installing docutils (0.16)
  • Installing email-validator (1.1.1)
  • Installing flask-babel (1.0.0)
  • Installing flask-jwt-extended (3.24.1)
  • Installing flask-login (0.4.1)
  • Installing flask-openid (1.2.5)
  • Installing flask-sqlalchemy (2.4.4)
  • Installing flask-wtf (0.14.3)
  • Installing jsonschema (3.2.0)
  • Installing lockfile (0.12.2)
  • Installing mako (1.1.3)
  • Installing marshmallow-enum (1.5.1)
  • Installing marshmallow-sqlalchemy (0.23.1)
  • Installing natsort (7.0.1)
  • Installing numpy (1.19.2)
  • Installing prison (0.1.3)
  • Installing python-dateutil (2.8.1)
  • Installing python-editor (1.0.4)
  • Installing python-slugify (4.0.1)
  • Installing pytzdata (2020.1)
  • Installing pyyaml (5.3.1)
  • Installing sqlalchemy-utils (0.36.8)
  • Installing tzlocal (1.5.1)
  • Installing urllib3 (1.25.10)
  • Installing alembic (1.4.3): Pending...
  • Installing argcomplete (1.12.1): Installing...
  • Installing cached-property (1.5.2): Pending...
  • Installing cattrs (1.0.0): Installing...
  • Installing colorlog (4.0.2): Downloading... 0%
  • Installing configparser (3.5.3): Pending...
  • Installing alembic (1.4.3)
  • Installing argcomplete (1.12.1)
  • Installing cached-property (1.5.2)
  • Installing cattrs (1.0.0)
  • Installing colorlog (4.0.2)
  • Installing configparser (3.5.3)
  • Installing croniter (0.3.35)
  • Installing dill (0.3.2)
  • Installing flask-admin (1.5.4)
  • Installing flask-appbuilder (2.3.4)
  • Installing flask-caching (1.3.3)
  • Installing flask-swagger (0.2.14)
  • Installing funcsigs (1.0.2)
  • Installing future (0.18.2)
  • Installing graphviz (0.14.2)
  • Installing gunicorn (20.0.4)
  • Installing iso8601 (0.1.13)
  • Installing json-merge-patch (0.2)
  • Installing lazy-object-proxy (1.5.1)
  • Installing markdown (2.6.11)
  • Installing pandas (1.1.3)
  • Installing pendulum (1.4.4)
  • Installing psutil (5.7.2)
  • Installing pygments (2.7.1)
  • Installing python-daemon (2.2.4)
  • Installing python-nvd3 (0.15.0)
  • Installing requests (2.24.0)
  • Installing setproctitle (1.1.10)
  • Installing sqlalchemy-jsonfield (0.9.0)
  • Installing tabulate (0.8.7)
  • Installing tenacity (4.12.0)
  • Installing thrift (0.13.0)
  • Installing unicodecsv (0.14.1)
  • Installing zope.deprecation (4.4.0)
  • Installing apache-airflow (1.10.12)

$ poetry run pip freeze | grep -i airflow
apache-airflow @ file:///Users/sujaymansingh/Library/Caches/pypoetry/artifacts/e7/ff/59/5a0135f91c9c796180c8d4fc525995da4da26564024fd75cf18e568874/apache_airflow-1.10.12-py2.py3-none-any.whl
$ poetry run pip freeze | grep -i celery
$ poetry run pip freeze | grep -i flower
$
```

However, installing that with virtualenv and pip ensures that all the dependencies are installed.

```
$ rm -rf virtualenv
$ python3 -m venv virtualenv
$ ./virtualenv/bin/pip install 'apache-airflow[celery]' -q
You are using pip version 19.0.3, however version 20.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
$ ./virtualenv/bin/pip freeze | grep -i airflow
apache-airflow==1.10.12
$ ./virtualenv/bin/pip freeze | grep -i celery
celery==4.4.7
$ ./virtualenv/bin/pip freeze | grep -i flower
flower==0.9.5
```

**Comment by abn** [comment]
@sujaymansingh can you please try with `1.1.3`? You might want to either perform a `poetry update` or `poetry lock --no-update` prior to testing.

**Comment by sujaymansingh** [comment]
Looks like using `poetry==1.1.3` fixes the issue, thanks @abn

**Comment by sujaymansingh** [comment]
Do you want me to close the issue?

**Comment by abn** [comment]
Closing as this was resolved with `1.1.3`.
